src/main/java/javaslang/collection/Map.java

```diff
@@ -6,16 +6,78 @@
 package javaslang.collection;
 
 import java.io.Serializable;
+import java.util.Iterator;
 import java.util.Objects;
 import java.util.function.Function;
 
 interface Map<K, V> extends /*TODO: Traversable<Map.Entry<K, V>>,*/ Function<K, V> {
 
+    static <K, V> Map<K, V> empty() {
+        return HashMap.empty();
+    }
+
+    /**
+     * Returns a singleton {@code List}, i.e. a {@code List} of one element.
+     *
+     * @param entry A map entry.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given entry
+     */
+    static <K, V> Map<K, V> of(Entry<? extends K, ? extends V> entry) {
+        final HashMap<K, V> map = HashMap.empty();
+        return map.put(entry.key, entry.value);
+    }
+
+    /**
+     * Creates a Map of the given entries.
+     *
+     * @param entries Map entries
+     * @param <K>     The key type
+     * @param <V>     The value type
+     * @return A new Map containing the given entries
+     */
+    @SafeVarargs
+    static <K, V> Map<K, V> of(Entry<? extends K, ? extends V>... entries) {
+        Objects.requireNonNull(entries, "entries is null");
+        HashMap<K, V> map = HashMap.empty();
+        for (Entry<? extends K, ? extends V> entry : entries) {
+            map = map.put(entry.key, entry.value);
+        }
+        return map;
+    }
+
+    /**
+     * Creates a Map of the given entries.
+     *
+     * @param entries Map entries
+     * @param <K>     The key type
+     * @param <V>     The value type
+     * @return A new Map containing the given entries
+     */
+    @SuppressWarnings("unchecked")
+    static <K, V> Map<K, V> ofAll(Iterable<? extends Entry<? extends K, ? extends V>> entries) {
+        Objects.requireNonNull(entries, "entries is null");
+        if (entries instanceof HashMap) {
+            return (HashMap<K, V>) entries;
+        } else {
+            HashMap<K, V> map = HashMap.empty();
+            for (Entry<? extends K, ? extends V> entry : entries) {
+                map = map.put(entry.key, entry.value);
+            }
+            return map;
+        }
+    }
+
```

Comment:
Please let's have static factory methods in _terminal collection interfaces/classes_ (like List, Stream, HashMap, ...) which are like classes. In the image these are the leafs.

![classes-overview](https://cloud.githubusercontent.com/assets/743833/9027154/491a0d3e-394a-11e5-8568-39f6db840b8d.png)

I'm not sure, if it helps to have in other interfaces static factory methods. My experience is, that it feels a little bit magic, when users are not familiar with the collection lib and don't know which instance they get. It is better to be explicit:

```java
Map<?, ?> map = HashMap.empty().put("Hi", "Ruslan");
```